mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICALToJsonAttribute.java

```diff
@@ -0,0 +1,164 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.transport.mailets;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import javax.mail.MessagingException;
+
+import org.apache.james.transport.mailets.model.ICAL;
+import org.apache.mailet.Mail;
+import org.apache.mailet.MailAddress;
+import org.apache.mailet.base.GenericMailet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.github.steveash.guavate.Guavate;
+import com.google.common.base.Strings;
+
+import net.fortuna.ical4j.model.Calendar;
+
+/**
+ * ICALToJsonAttribute takes a map of ICAL4J objects attached as attribute, and output the list of corresponding json as
+ * an other attribute.
+ *
+ * The JSON contains the following fields :
+ *
+ * <ul>
+ *     <li><b>ical</b> : the raw ical string, in UTF-8</li>
+ *     <li><b>sender</b> : the sender of the mail (compulsory, mail without sender will be discarded)</li>
+ *     <li><b>recipient</b> : the recipient of the mail. If the mail have several recipients, each recipient will have
+ *     its own JSON.</li>
+ *     <li><b>uid</b> : the UID of the ical (optional)</li>
+ *     <li><b>sequence</b> : the sequence of the ical (optional)</li>
```

Comment:
If your question is why it's optional. I think we can make it non optional because the RFC state the default value is 0 so if it's not inside the ICS it mean it's zero.

>    Description:  When a calendar component is created, its sequence
>       number is 0.  It is monotonically incremented by the "Organizer's"
>       CUA each time the "Organizer" makes a significant revision to the
>       calendar component.